faiss/python/__init__.py

```diff
@@ -211,6 +211,10 @@ def replacement_add(self, x):
         """
 
         n, d = x.shape
+
+        if x.dtype != np.float32:
+            raise TypeError("Input argument %s must be ndarray of dtype float32, but found %s" % ("x", str(x.dtype)))
```

Comment:
Please make sure that the line is < 80 characters.
Nit: %s already applies str() on its argument, so str() is not necessary.

Reply:
now addressed